cmd/kubeadm/app/master/addons.go

```diff
@@ -216,16 +214,11 @@ func createKubeDNSPodSpec(cfg *kubeadmapi.MasterConfiguration) v1.PodSpec {
 					SuccessThreshold:    1,
 					FailureThreshold:    5,
 				},
-				// The code below is a workaround for https://github.com/kubernetes/contrib/blob/master/dnsmasq-metrics/Dockerfile.in#L21
-				// This is just the normal mode (to run with user 0), all other containers do it except for this one, which may lead to
-				// that the DNS pod fails if the "nobody" _group_ doesn't exist. I think it's a typo in the Dockerfile manifest and
-				// that it should be "USER nobody:nogroup" instead of "USER nobody:nobody". However, this fixes the problem.
-				SecurityContext: &v1.SecurityContext{
```

Comment:
I think this needs to stay unfortunately, see here: https://github.com/kubernetes/dns/blob/master/Dockerfile.sidecar#L21

It broke on my machines; I think it should be `USER nobody:nogroup` instead of `USER nobody:nobody`. For me, the `nobody` didn't exist and the whole dns thing was broken.